Unit/e2e tests. UI Fixes.

DoctorBud opened the pull request:
- Adds e2e test for node.
- Gets e2e and unit tests working again.
- Fixes several bugs.
- Rename 'publish' to 'ghpublish' to avoid confusion with normal 'npm
  publish'.

nathandunn commented on src/components/GenomeFeature.vue:
```diff
@@ -32,13 +32,14 @@ export default {
     },
   },
   data() {
+    console.log(this.mygeneData);
```
remove unless you need it here

nathandunn commented on src/api/BioLink.js:
```diff
@@ -393,6 +443,8 @@ export function getNodeAssociations(nodeType, identifier, cardType, params) {
           reject(responseData);
         }
         else {
+          // console.log('getNodeAssociations', nodeType, identifier, cardType, url);
```
clean out

DoctorBud replied:
Until BL stabilizes, I'm often debugging its response and these console.log are worth having around, so I try to keep them commented out, but in the code.

nathandunn commented on src/api/BioLink.js:
```diff
@@ -213,10 +213,39 @@ async function getCountsForNode(nodeId, nodeType) {
+  };
+  const bioentityResp = await axios.get(bioentityUrl, { params: bioentityParams });
+  const bioentityResponseData = bioentityResp.data;
+  console.log(bioentityResp.request.responseURL);
```
remove loggin unless not needed

DoctorBud replied:
This logging may be necessary if BL's `get_association_counts` gets fixed and I need to test the new version of getCounts above. Once I'm confident the BL endpoint works, I'll remove the old count code and all the logging. Meanwhile, I'm leaving it in.

DoctorBud replied:
https://github.com/monarch-initiative/monarch-ui/pull/40/files/e965086cb5924a2386508ae246f5c6757874ddac#diff-ec1bbcbf802856ad9001d1b8c3b89ce3R296

nathandunn commented on src/api/BioLink.js:
```diff
@@ -113,9 +113,9 @@ const nodeAssociationTypes = {
-    'model',
-    'variant',
-    'literature'
+    // 'model',
```
+1 to annotating thiese

DoctorBud replied:
Biolink is missing genotype-to-model and genotype-to-variant associations right now. Not sure if genotype-to-literature will ever be supported.

nathandunn commented on tests/unit/test-utils.js:
```diff
@@ -0,0 +1,14 @@
+  return shallowMount(component, newOptions);
+}
+
+// const index = {
```
remove?

nathandunn commented:
E2E tests:

    npm run-script test:e2e 

test e2e logs:

```

> monarch-ui@0.0.2 test:e2e /Users/nathandunn/repositories/monarch-ui
> vue-cli-service test:e2e

 INFO  Starting development server...
WARNING: Invalid value for $grid-breakpoints: This map must be in ascending order, but key 'monarch' has value 860px which isn't greater than 1200px, the value of the previous key 'xl' !
         on line 16 of node_modules/bootstrap/scss/_functions.scss, in mixin `-assert-ascending`
         from line 201 of node_modules/bootstrap/scss/_variables.scss
         from line 9 of node_modules/bootstrap/scss/bootstrap.scss
         from line 26 of stdin



 WARNING  Compiled with 2 warnings                                                                                                                                              13:30:41

 warning  

asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets: 
  img/ingest_curation_workflow.f2c3456f.png (435 KiB)
  img/fontawesome-webfont.912ec66d.svg (434 KiB)
  img/infinity.82e3332f.gif (306 KiB)
  js/chunk-vendors.7dcf2ea4.js (1.06 MiB)
  img/team/AnneThessen.png (833 KiB)
  img/team/CraigMcNamara.jpg (462 KiB)
  img/team/EdwinZhang.jpg (447 KiB)
  img/team/TomConlin.png (524 KiB)
  img/team/ValentinaCipriani.jpg (320 KiB)

 warning  

entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  app (1.5 MiB)
      css/chunk-vendors.2606051e.css
      js/chunk-vendors.7dcf2ea4.js
      css/app.c6e7e043.css
      js/app.b3442c09.js



  App running at:
  - Local:   http://localhost:8080/ 
  - Network: http://172.16.1.120:8080/

  App is served in production mode.
  Note this is for preview or E2E testing only.

Starting selenium server... started - PID:  32391

[Test] Test Suite
=====================

Running:  default e2e tests
 ✔ Element <#app> was visible after 171 milliseconds.
 ✖ Testing if element <#monarch-home-container> is present.  - expected "present" but got: "not present"
    at Object.default e2e tests (/Users/nathandunn/repositories/monarch-ui/tests/e2e/specs/test.js:9:15)
    at _combinedTickCallback (internal/process/next_tick.js:132:7)


FAILED:  1 assertions failed and 1 passed (6.184s)

 _________________________________________________

 TEST FAILURE:  1 assertions failed, 1 passed. (6.291s)

 ✖ test

   - default e2e tests (6.184s)
   Testing if element <#monarch-home-container> is present.  - expected "present" but got: "not present"
       at Object.default e2e tests (/Users/nathandunn/repositories/monarch-ui/tests/e2e/specs/test.js:9:15)
       at _combinedTickCallback (internal/process/next_tick.js:132:7)

 ERROR  Error: Command failed: /Users/nathandunn/repositories/monarch-ui/node_modules/nightwatch/bin/nightwatch --config /Users/nathandunn/repositories/monarch-ui/node_modules/@vue/cli-plugin-e2e-nightwatch/nightwatch.config.js --env chrome
Error: Command failed: /Users/nathandunn/repositories/monarch-ui/node_modules/nightwatch/bin/nightwatch --config /Users/nathandunn/repositories/monarch-ui/node_modules/@vue/cli-plugin-e2e-nightwatch/nightwatch.config.js --env chrome
    at makeError (/Users/nathandunn/repositories/monarch-ui/node_modules/execa/index.js:174:9)
    at Promise.all.then.arr (/Users/nathandunn/repositories/monarch-ui/node_modules/execa/index.js:278:16)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! monarch-ui@0.0.2 test:e2e: `vue-cli-service test:e2e`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the monarch-ui@0.0.2 test:e2e script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/nathandunn/.npm/_logs/2019-02-04T21_30_49_204Z-debug.log
```

nathandunn commented:
E2E logging:

```
0 info it worked if it ends with ok
1 verbose cli [ '/Users/nathandunn/.nvm/versions/node/v8.12.0/bin/node',
1 verbose cli   '/Users/nathandunn/.nvm/versions/node/v8.12.0/bin/npm',
1 verbose cli   'run-script',
1 verbose cli   'test:e2e' ]
2 info using npm@6.5.0
3 info using node@v8.12.0
4 verbose run-script [ 'pretest:e2e', 'test:e2e', 'posttest:e2e' ]
5 info lifecycle monarch-ui@0.0.2~pretest:e2e: monarch-ui@0.0.2
6 info lifecycle monarch-ui@0.0.2~test:e2e: monarch-ui@0.0.2
7 verbose lifecycle monarch-ui@0.0.2~test:e2e: unsafe-perm in lifecycle true
8 verbose lifecycle monarch-ui@0.0.2~test:e2e: PATH: /Users/nathandunn/.nvm/versions/node/v8.12.0/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/Users/nathandunn/repositories/monarch-ui/node_modules/.bin:/usr/local/neo4j/bin:/Users/nathandunn/.nvm/versions/node/v8.12.0/bin:/usr/local/sbin:/Users/nathandunn/.sdkman/candidates/micronaut/current/bin:/Users/nathandunn/.sdkman/candidates/groovy/current/bin:/Users/nathandunn/.sdkman/candidates/grails/current/bin:/Users/nathandunn/.sdkman/candidates/gradle/current/bin:/usr/local/opt/coreutils/libexec/gnubin:/usr/local/bin:/Library/Java/JavaVirtualMachines/jdk1.8.0_162.jdk/Contents/Home/bin:/usr/sbin:/usr/local/maven/bin:/Users/nathandunn/tomcat/bin:/usr/local/bin:/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/Users/nathandunn/pear/bin:/opt/local/lib/postgresql83/bin:/Users/nathandunn/Downloads/atlassian-plugin-sdk-3/bin:/usr/local/tools:/bin:/usr/local/ant/bin:/Users/nathandunn/bin:/Users/nathandunn/tomcat/bin:/opt/nginx/sbin:/Users/nathandunn/Downloads/srilm/bin/macosx-m64:/usr/local/launch4j
9 verbose lifecycle monarch-ui@0.0.2~test:e2e: CWD: /Users/nathandunn/repositories/monarch-ui
10 silly lifecycle monarch-ui@0.0.2~test:e2e: Args: [ '-c', 'vue-cli-service test:e2e' ]
11 silly lifecycle monarch-ui@0.0.2~test:e2e: Returned: code: 1  signal: null
12 info lifecycle monarch-ui@0.0.2~test:e2e: Failed to exec test:e2e script
13 verbose stack Error: monarch-ui@0.0.2 test:e2e: `vue-cli-service test:e2e`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/Users/nathandunn/.nvm/versions/node/v8.12.0/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
13 verbose stack     at emitTwo (events.js:126:13)
13 verbose stack     at EventEmitter.emit (events.js:214:7)
13 verbose stack     at ChildProcess.<anonymous> (/Users/nathandunn/.nvm/versions/node/v8.12.0/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at emitTwo (events.js:126:13)
13 verbose stack     at ChildProcess.emit (events.js:214:7)
13 verbose stack     at maybeClose (internal/child_process.js:915:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
14 verbose pkgid monarch-ui@0.0.2
15 verbose cwd /Users/nathandunn/repositories/monarch-ui
16 verbose Darwin 18.2.0
17 verbose argv "/Users/nathandunn/.nvm/versions/node/v8.12.0/bin/node" "/Users/nathandunn/.nvm/versions/node/v8.12.0/bin/npm" "run-script" "test:e2e"
18 verbose node v8.12.0
19 verbose npm  v6.5.0
20 error code ELIFECYCLE
21 error errno 1
22 error monarch-ui@0.0.2 test:e2e: `vue-cli-service test:e2e`
22 error Exit status 1
23 error Failed at the monarch-ui@0.0.2 test:e2e script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

nathandunn commented:
It occurs to me I had not switched your branch when I started testing

Everything works (tests passed, etc.).  Sorry for the confusion.

DoctorBud commented:
I just updated package.json and added package-lock.json. @nathandunn please download my latest update and do an `rm -rf node_modules/ && npm install` and the unit tests should work properly. It was due to a bug in the `prettier` module which I believe has been fixed.

I'm still not sure why e2e is failing on you. We may need to screenshare to debug it.
